lib/irc/BridgedClient.js

```diff
@@ -77,6 +77,14 @@ BridgedClient.prototype.getClientConfig = function() {
     return this._clientConfig;
 };
 
+BridgedClient.prototype.kill = function() {
+    // nullify references to connection instance client
```

Comment:
Option number 2. I didn't consider GCing because at the moment, the only reason for a client to be killed is if the entire bridge is killed. I'll change the comment :+1: 

Also, ```unsafeClient``` is mostly null guarded but sometimes not explicitly, with some functions relying on calling another and it succeeding and I think that may be a separate problem.